# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/WSRunObject.cpp
# Commit: 310cd44cff95
# Full Hash: 310cd44cff95c94b9982619c5fc205c63b56352b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 04:34:13
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 4-2: Make `HTMLEditor::ChangeListElementType` stop touching `Selection` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152967
# ==============================================================================

diff -r d61ac9d43588 -r 310cd44cff95 editor/libeditor/WSRunObject.cpp
--- a/editor/libeditor/WSRunObject.cpp	Thu Aug 04 01:50:28 2022 +0000
+++ b/editor/libeditor/WSRunObject.cpp	Thu Aug 04 02:02:46 2022 +0000
@@ -656,6 +656,9 @@
       if (NS_WARN_IF(convertListTypeResult.EditorDestroyed())) {
         return EditActionResult(NS_ERROR_EDITOR_DESTROYED);
       }
+      // There is AutoTransactionConserveSelection above, therefore, we don't
+      // need to update selection here.
+      convertListTypeResult.IgnoreCaretPointSuggestion();
       NS_WARNING_ASSERTION(
           convertListTypeResult.isOk(),
           "HTMLEditor::ChangeListElementType() failed, but ignored");
